src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1007,9 +1045,367 @@ private boolean readBoolean_1_1() {
+        if (marker.startIndex < 0) {
+            // TODO this is temporary until the Ion 1.1 system symbol table is finalized. At that point, we will
+            //  look up the symbol ID (held in `marker.endIndex`) in the system symbol table. Below, 10 is the
+            //  number of Ion 1.1 system symbols, providing the conversion from local symbol ID to `symbols` array
+            //  index.
+            return ION_ENCODING.equals(symbols[(int) (marker.endIndex) - 10]);
+        } else {
+            return bytesMatch(ION_ENCODING_UTF8, buffer, (int) marker.startIndex, (int) marker.endIndex);
+        }
+    }
+
+    /**
+     * @return true if the reader is positioned on an encoding directive; otherwise, false.
+     */
+    private boolean isPositionedOnEncodingDirective() {
+        return event == Event.START_CONTAINER
+            && hasAnnotations
+            && valueTid.type == IonType.SEXP
+            && parent == null
+            && startsWithIonEncoding();
+    }
+
+    /**
+     * Grows the `symbols` array to the next power of 2 that will fit the current need.
+     */
+    protected void growSymbolsArray(int shortfall) {
+        int newSize = nextPowerOfTwo(symbols.length + shortfall);
+        String[] resized = new String[newSize];
+        System.arraycopy(symbols, 0, resized, 0, localSymbolMaxOffset + 1);
+        symbols = resized;
+    }
+
+    private void resetSymbolTable() {
+        // The following line is not required for correctness, but it frees the references to the old symbols,
+        // potentially allowing them to be garbage collected.
+        Arrays.fill(symbols, 0, localSymbolMaxOffset + 1, null);
+        localSymbolMaxOffset = -1;
+    }
+
+    /**
+     * Installs the given symbols at the end of the `symbols` array.
+     * @param newSymbols the symbols to install.
+     */
+    protected void installSymbols(List<String> newSymbols) {
+        if (newSymbols != null) {
+            int numberOfNewSymbols = newSymbols.size();
+            int numberOfAvailableSlots = symbols.length - (localSymbolMaxOffset + 1);
+            int shortfall = numberOfNewSymbols - numberOfAvailableSlots;
+            if (shortfall > 0) {
+                growSymbolsArray(shortfall);
+            }
+            int i = localSymbolMaxOffset;
+            for (String newSymbol : newSymbols) {
+                symbols[++i] = newSymbol;
+            }
+            localSymbolMaxOffset += newSymbols.size();
+        }
+    }
+
+    /**
+     * @return the {@link EncodingContext} currently active, or {@code null}.
+     */
+    EncodingContext getEncodingContext() {
+        return macroEvaluator == null ? null : macroEvaluator.getEncodingContext();
+    }
+
+    /**
+     * Reads encoding directives from the stream. Capable of resuming if not enough data is currently available to
+     * complete the encoding directive.
+     */
+    private class EncodingDirectiveReader {
```

Comment:
See the SymbolTableReader in IonReaderContinuableApplicationBinary for another example of this pattern. The state machine allows the reader to resume reading the encoding directive even if the stream ends in the middle of it.